### ISSUE TITLE
Data-driven system definition

### ISSUE THREAD

**Issue by weavejester** [duct-framework/duct#41]
Currently the system is created through the `component/system-map` function:

```clojure
(defn new-system [config]
  (let [config (meta-merge base-config config)]
    (-> (component/system-map
         :app     (handler-component (:app config))
         :http    (jetty-server (:http config))
         :example (endpoint-component example-endpoint))
        (component/system-using
         {:http [:app]
          :app  [:example]}))))
```

But instead we could define the system via a data structure:

```edn
{:components
 {:app  duct.component.handler/handler-component
  :http ring.component.jetty/jetty-server}
 :endpoints
 {:example bar.endpoint.example/example-endpoint}
 :dependencies
 {:http [:app]
  :app  [:example]}}
```

And then have a `system-factory` function to construct the `new-system` function from the data structure.

```clojure
(def new-system
  (system-factory (read-edn "foo/system.edn")))
```

As well as reducing the amount of code, it would also allow generators to more easily add components and endpoints to the system.

**Comment by adairdavid** [comment]
Is there any nuance to incorporating *other libraries'* components into the new edn-based configuration? Since these libraries must be `require`'d at some point, wouldn't their vars return ClassNotFound exceptions on use?

I'm trying to incorporate `fixtures-component`, but I receive errors that lead me to indicate that the vars I place in the configuration map lead to ClassNotFound Exceptions. I've created an issue over at `fixtures-component` with more information and stacktraces: https://github.com/banzai-inc/fixtures-component/issues/3

My overall question, is how can 3rd party components be incorporated into the configuration map? I've successfully incorporated `duct` components, and my own created ones, but there seems to be a nuance to using 3rd party libraries that haven't been `require`'d. Unless this is an error specific to `fixtures-component`.
